mlir/lib/Dialect/Vector/Transforms/VectorLinearize.cpp

```diff
@@ -265,10 +266,11 @@ struct LinearizeVectorShuffle final
   matchAndRewrite(vector::ShuffleOp shuffleOp, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
     Type dstType = getTypeConverter()->convertType(shuffleOp.getType());
-    assert(!(shuffleOp.getV1VectorType().isScalable() ||
-             shuffleOp.getV2VectorType().isScalable() ||
-             cast<VectorType>(dstType).isScalable()) &&
-           "scalable vectors are not supported.");
+    if (shuffleOp.getV1VectorType().isScalable() ||
```

Comment:
`vector.shuffle` does not support scalable vectors, so keeping an assert should be fine for this one:
* https://mlir.llvm.org/docs/Dialects/Vector/#vectorshuffle-vectorshuffleop

A comment explaining the rationale for using `assert` rather than `notifyMatchFailure` would be welcome :)

Reply:
+1 on test coverage.

Reply:
Should be addressed now